core/Azure.Mcp.Core/src/Commands/ToolMetadata.cs

```diff
@@ -111,8 +192,24 @@ public sealed class ToolMetadata
     /// The default is <see langword="false"/>.
     /// </para>
     /// </remarks>
+    [JsonIgnore]
+    public bool LocalRequired
+    {
+        get => _localRequired;
+        init => _localRequired = value;
+    }
+
+    /// <summary>
+    /// Gets the localRequired metadata property with value and description for serialization.
+    /// </summary>
     [JsonPropertyName("localRequired")]
-    public bool LocalRequired { get; init; } = false;
+    public MetadataDefinition LocalRequiredProperty => _localRequiredProperty ??= new MetadataDefinition
+    {
+        Value = _localRequired,
+        Description = _localRequired
+            ? "This tool requires local execution environment or local resources to function properly."
```

Comment:
"This tool is only available when the Azure MCP server is configured to run as a Local MCP Server (STDIO)."